src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -262,19 +220,11 @@ private async Task ProcessQueueAsync()
 
                 // If there was an in progress work item that failed in the queue logic, set the result of the queue item
                 // to the exception so that it bubbles back to the caller.
-                inProgressWorkItem?.HandleQueueFailure(e);
+                inProgressWorkItem?.TrySetException(e);
                 OnRequestServerShutdown($"Error occurred processing queue in {_serverName}: {e.Message}.");
             }
         }
 
-        private static Task ExecuteCallbackAsync(QueueItem work, RequestContext? context, CancellationToken queueCancellationToken)
-        {
-            // Create a combined cancellation token to cancel any requests in progress when this shuts down
-            using var combinedTokenSource = queueCancellationToken.CombineWith(work.CancellationToken);
```

Comment:
handled in the work.CallbackAsync method now - the queue item is responsible for the cancellation token from the client and the queue cancellation token is passed in.  Before handling the request the tokens are combined and checked for cancellation

Reply:
note: an alternative formalizatino woudl be to *not* store hte combined token in the item, and instead have a queue of `(IQueueItem, CancellationToken)`.  They're isomorphic.  But it's a question of who should morally own the CT.